models/Element/Tag/Dao.php

```diff
@@ -121,14 +121,15 @@ public function delete(): array
      */
     public function getTagsForElement(string $cType, int $cId): array
     {
-        $tags = [];
         $tagIds = $this->db->fetchFirstColumn('SELECT tagid FROM tags_assignment WHERE cid = ? AND ctype = ?', [$cId, $cType]);
-
-        foreach ($tagIds as $tagId) {
-            $tags[] = Model\Element\Tag::getById($tagId);
+        if (empty($tagIds)) {
+            return [];
         }
 
-        $tags = array_filter($tags);
+        $list = new Listing();
+        $list->setCondition("`id` IN ('" . implode("','", $tagIds) . "')");
```

Comment:
should we use `$db->quote` here just to be safe?

Reply:
For the tag IDs? They come directly from an integer column in the database, so it should be safe. 

But if you want to escape them, we should probably use prepared statements here as well:
```suggestion
        $list->setCondition("`id` IN (?)", [$tagIds]);
```